main.cpp

```diff
@@ -439,6 +386,87 @@ IntType& IntType::divide( int rhs )
     return *this;
 }
 
+IntType& IntType::pow( int rhs)
+{    
+    return powInternal(rhs);
+}
+
+IntType& IntType::pow(const FloatType& argType)
+{    
+    return powInternal(static_cast<int>(argType));
+}
+
+IntType& IntType::pow(const DoubleType& argType)
+{    
+    return powInternal(static_cast<int>(argType));
+}
+
+IntType& IntType::pow(const IntType& argType)
+{    
+    return powInternal(static_cast<int>(argType));
+}
+
+IntType& IntType::powInternal(const int arg)
+{
+    *value = static_cast<int>(std::pow( *value, arg ));
+    return *this;
+}
+
+struct Point
+{
+    Point(float a, float b) : x(a), y(b) {}
+
+    Point(const FloatType& a, const FloatType& b) :
+    x(static_cast<float>(a)),
+    y(static_cast<float>(b)) {}
+
+    Point(const IntType& a, const IntType& b) :
+    x(static_cast<int>(a)),
+    y(static_cast<int>(b)) {}
+
+    Point(const DoubleType& a, const DoubleType& b) :
+    x(static_cast<float>(a)),
+    y(static_cast<float>(b)) {}
```

Comment:
Demonstrate constructor forwarding in order to follow D.R.Y. principles:
call the constructor that takes 2 floats from each of these constructors.
```
T(float, float) { }
T(const U& u1, const U& u2) : T( cast(u1), cast(u2) ) { }
```
- [ ] revise